app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,17 @@
+class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  def openid_connect
+    @user = User.from_omniauth(request.env["omniauth.auth"])
+    if @user.persisted?
+      sign_in_and_redirect @user
+      set_flash_message(:notice, :success, kind: "OpenID Connect") if is_navigational_format?
```

Comment:
nit: these lines are becoming really long, I had to zoom out on my machine to see until the end. Maybe we could use early return instead of big if/else chains